### DISCUSSION TITLE
plotting unstructured grids with add_volume method is abnormally slow

### DISCUSSION THREAD

**Discussion by Autmlif** [pyvista/pyvista#5035] [Q&A]
Hi,
when I use add_volume to draw my unstructured grids, the speed of generating or dragging plot is unbearably slow.
I can't figure out the reason.
Here is my data info.
![image](https://github.com/pyvista/pyvista/assets/65207622/09532e10-4e36-43db-8597-469efe7ddc5b)

relating code ↓

![image](https://github.com/pyvista/pyvista/assets/65207622/99d1262e-b6c0-4605-935d-32d1dee63f44)

I wonder if there are some interpolation algorithms to make the data smoother and thus rendering faster?

**Comment by banesullivan** [comment]
This is a known problem for volume rendering unstructured grids in VTK (we have a warning in the documentation about this 😉): https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.add_volume.html#pyvista.Plotter.add_volume

In my experience, I typically get better performance and results by resampling the UnstructuredGrid to ImageData then volume rendering that
